geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureConverterFactory.scala

```diff
@@ -115,6 +130,35 @@ trait ToSimpleFeatureConverter[I] extends SimpleFeatureConverter[I] with LazyLog
   def inputFields: IndexedSeq[Field]
   def idBuilder: Expr
   def fromInputType(i: I): Seq[Array[Any]]
+  def validating: Boolean
+
+  val validate: (SimpleFeature, EvaluationContext) => Boolean = {
+    val valid: (SimpleFeature) => Boolean = {
+      import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+      val dtgFn: (SimpleFeature) => Boolean = targetSFT.getDtgIndex match {
+        case Some(dtgIdx) => (sf: SimpleFeature) => sf.getDefaultGeometry != null && sf.getAttribute(dtgIdx) != null
```

Comment:
I think ideally we should be able to handle features without dates in our indexing (even if the feature type has a date)... but this is good for now.